specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json

```diff
@@ -0,0 +1,157 @@
+{
+    "swagger": "2.0",
+    "info": {
+      "version": "2018-11-01-preview",
+      "title": "BillingManagementClient",
+      "description": "Billing client provides access to billing resources for Azure subscriptions."
+    },
+    "host": "management.azure.com",
+    "schemes": [
+      "https"
+    ],
+    "consumes": [
+      "application/json"
+    ],
+    "produces": [
+      "application/json"
+    ],
+    "paths": {
+      "/providers/Microsoft.Subscription/subscriptions/{subscriptionId}/cancel": {
```

Comment:
>cancel [](start = 72, length = 6)

How long does cancelling a subscription take. Should it be a long running operation?

Reply:
Also, the API seems a bit asymmetric. How do you "un-cancel" a subscription?

---
In reply to: [257442715](https://github.com/Azure/azure-rest-api-specs/pull/5209#discussion_r257442715) [](ancestors = 257442715)

Reply:
This API calls another internal API, which returns back the response with new subscription Id in 1-2 seconds and that internal API, after returning back response to us, calls other services which provision subscriptions which can take further couple of seconds. Right now, we are not providing functionality to the customers to re-enable the subscription.